common/archiver/provider/provider.go

```diff
@@ -45,9 +47,9 @@ type (
 			serviceName string,
 			historyContainer *archiver.HistoryBootstrapContainer,
 			visibilityContainter *archiver.VisibilityBootstrapContainer,
-		)
-		GetHistoryArchiver(scheme string, serviceName string) (archiver.HistoryArchiver, error)
-		GetVisibilityArchiver(scheme string, serviceName string) (archiver.VisibilityArchiver, error)
+		) error
+		GetHistoryArchiver(URI archiver.URI, serviceName string) (archiver.HistoryArchiver, error)
```

Comment:
I think you should just take `scheme string` here. My earlier suggestion was that you could take `uri string` but that was before realized that URI had to live in archiver package. Now that URI belongs to archiver rather than provider I think provider should just take scheme.